src/google-maps/google-map/google-map.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  Input,
+  OnInit,
+  ViewChild,
+} from '@angular/core';
+import {ReplaySubject} from 'rxjs';
+import {take} from 'rxjs/operators';
+
+/**
+ * Angular component that renders a Google Map via the Google Maps JavaScript
+ * API.
+ * @see https://developers.google.com/maps/documentation/javascript/reference/
+ */
+@Component({
+  selector: 'google-map',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: '<div #map></div>',
+})
+export class GoogleMap implements OnInit {
+  @Input() height = '500px';
+  @Input() width = '500px';
+  // TODO(mbehrlich): add options, handlers, properties, and methods.
+
+  @ViewChild('map', {static: true}) set mapEl(mapEl: ElementRef) {
+    this._mapEl$.next(mapEl);
```

Comment:
Why make this a stream? It seem like this element will never change, no? You could just use `ngAfterViewInit` to know that the query is resolved, but honestly, it's more straightforward to just do a `querySelector` than to depend on Angular when all you need is the native element. So I would just omit the `@ViewChild` and do the `querySelector` directly in `ngOnInit`.

Reply:
The original reason behind making this a stream was to force Angular to insert the map into the DOM before changes to the view occur. I didn't want to use ngAfterViewInit, because it was my understanding that Angular forbids changes to the view after it has been composed. I also don't want to use a querySelector because I am worried about selecting an element outside of the component.

I have since found out that using static (which I already am) with ViewChild allows you to use it in the ngOnInit lifecycle hook, so I think the best solution is to get rid of the ReplaySubject, but retain the ViewChild.

Reply:
For `querySelector`, I meant specifically injecting `ElementRef` and doing `querySelector` on that, not the full `document.querySelector`. The reason I'd suggest avoiding the Angular query stuff is that ti adds additional overhead (in terms of generated code and run-time, albeit minor), that isn't really necessary if you only need to perform DOM operations and don't need to access any angular constructs.

Reply:
Ah, I see. Done.